### PR TITLE
Note on polymorphic contracts and recursive records

### PR DESCRIPTION
**Pull request by yannham**
cf https://github.com/tweag/nickel/issues/1161#issuecomment-1462397141

### CODE REVIEW

**Comment by yannham** [comment]
cc @vkleen

File: notes/fixing-sealing-and-recursive-records.md [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+up being paired one to one in program being evaluated. However, `%seal%` and
+`%unseal%` are generated in different places.
+
+If a contract containing an as yet unpaired `%seal%` or `%unseal%` ends up being
+attached to a record field, any reference to this field will have to apply the
+contract and essentially duplicates the `%seal%` or `%unseal%` operation. This
+means that if a type containing a free type variable is turned into a contract,
+and this contract gets attached to a record field for some reason, any reference
+to the field has the potential to produce a mismatch between `%seal%` and
+`%unseal%` primops in the program.
```
Comment (yannham) [inline notes/fixing-sealing-and-recursive-records.md:25]: Might be useful to reproduce the original example here, just for the sake of (relative) self containment.

Comment (yannham) [reply notes/fixing-sealing-and-recursive-records.md:25]: The example is detailed below, though. Still, it's very drafty, and it's probably better if one can understand what this is about without having to dive into the second part if not needed.

**Review by yannham — COMMENTED** [review]
Thanks for taking care of this. I can't approve my own PR, but I'm fine with merging it in this state (though I left a review comment to be looked at)

**Review by vkleen — APPROVED** [review]

**Pull request merged**